src/fundus/scraping/common_crawl/html.py

```diff
@@ -0,0 +1,66 @@
+from typing import Dict, Iterator, List, Optional, cast
+from urllib.parse import urlparse
+
+import requests
+from fastwarc import ArchiveIterator, WarcRecord, WarcRecordType
+from ftfy import guess_bytes
+
+from fundus.logging import basic_logger
+from fundus.publishers.base_objects import PublisherEnum
+from fundus.scraping.filter import URLFilter
+from fundus.scraping.html import HTML, WarcSource
+
+
+class CCNewsSource:
+    def __init__(self, *publishers: PublisherEnum, warc_path: str, headers: Optional[Dict[str, str]] = None):
```

Comment:
Thanks for pointing this out. Some files, especially the HTML sections, lack a major rework. I can imagine this being somewhat poorly planned, I will wait until I can rewrite the async part before tackling those reworks.

1. Yes
2. You're right
3. The first one is to filter URLs specific per publisher, the second per crawl. Because the CCNewsSource takes into account all publishers, we do not need the first one.

Reply:
> I will wait until I can rewrite the async part before tackling those reworks.

Ok. Since this affects some other comments as well. Maybe it would be good to collect these comments in an issue so we do not forget these details.

> The first one is to filter URLs specific per publisher, the second per crawl. Because the CCNewsSource takes into account all publishers, we do not need the first one.

Thanks, that makes sense.